symplyphysics/core/vectors/arithmetics.py

```diff
@@ -194,3 +194,49 @@ def reject_cartesian_vector(
         original_vector_,
         scale_vector(-1, project_vector(original_vector_, target_vector_))
     )
+
+
+def diff_cartesian_vector_components(
+    vector_: Vector,
+    *symbols: Any,
+    **kwargs: Any,
+) -> Vector:
+    if vector_.coordinate_system.coord_system_type != CoordinateSystem.System.CARTESIAN:
+        raise ValueError("Component-wise vector differentiation is only supported for Cartesian coordinates")
```

Comment:
I've found some formulas but they don't work for the type of vectors in curvilinear coordinate that we have. This is actually a bigger issue I stumbled into not so long ago. I guess this deserves a separate discussion.

Reply:
Cool. You can prepare a draft for your proposal. Eventually we want to avoid cartesian specific functions. Having 4 dimensional general arithmetics is even better but enormously difficult to achieve. Also cross product does not work in higher dimensions.